gprofiler/profilers/java.py

```diff
@@ -676,11 +685,16 @@ def _check_async_profiler_loaded(self, process: Process) -> bool:
         return False
 
     def _profile_process(self, process: Process) -> Optional[StackToSampleCount]:
-        if not self._is_profiling_supported(process):
-            return None
+        comm = process_comm(process)
+
+        if self._safemode_disable_reason is not None:
+            return self._profiling_skipped_stack(f"disabled due to {self._safemode_disable_reason}", comm)
+
+        if not self._is_jvm_profiling_supported(process):
+            return self._profiling_skipped_stack("profiling this JVM is not supported", comm)
```

Comment:
consider splitting to the sub reasons from the function (jvm version/jvm type/process name)

Reply:
Not sure it's relevant. This is only a marker for the error - from that point you can inspect the log and see the exact version. In the stack itself - I think it is too much. So even if you see a stack "JVM version bla bla" or "JVM type bla bla" you still have to check the log :shrug:

Reply:
k